apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataClientTest.cs

```diff
@@ -120,13 +121,99 @@ public void GetProjectMetadata()
         }
 
         [Fact]
+        public async Task InstanceMetadataChanged()
+        {
+            const string key = "my_instance_key1";
+            const string newValue = "foo";
+
+            var client = MetadataClient.Create();
+            var originalValue = await client.GetCustomInstanceMetadataAsync(key);
+
+            var waitHandle = new ManualResetEventSlim(false);
+            int eventCount = 0;
+            client.InstanceMetadataChanged += (s, e) =>
+            {
+                eventCount++;
+                waitHandle.Set();
+            };
+
+            try
+            {
+                await _fixture.UpdateMetadataAsync($"instance/attributes/{key}", newValue);
+                waitHandle.Wait(TimeSpan.FromSeconds(10));
+                Assert.Equal(1, eventCount);
+                Assert.Equal(newValue, await client.GetCustomInstanceMetadataAsync(key));
+            }
+            finally
```

Comment:
This is to restore the server's data back to the original state so this test's run does not impact tests which run after it. I was also considering just having a way to tell the server to reset itself so I didn't need to cache and restore the original value, but I think regardless of how its done, something like this should be done.